Userland/Utilities/uniq.cpp

```diff
@@ -37,41 +38,104 @@ static FILE* get_stream(char const* filepath, char const* perms)
     return ret;
 }
 
+static void print_line(char* line, unsigned long long count, bool show_duplicates, bool print_count, FILE* outfile)
+{
+    if (!show_duplicates || count > 0) {
+        if (print_count)
+            fprintf(outfile, "%llu %s", count + 1, line);
+        else
+            fputs(line, outfile);
+    }
+}
+
+static ssize_t get_line(char** buf, size_t* len, FILE* infile)
+{
+    errno = 0;
+    ssize_t rc = getline(buf, len, infile);
+    if (rc < 0 && errno != 0) {
+        perror("getline");
+        exit(1);
+    }
+    return rc;
+}
+
+static StringView skip(char* buf, int len, unsigned int nchars, unsigned int nfields)
```

Comment:
Why is this passed as a `buf` and `len` rather than just a StringView which has both of those?